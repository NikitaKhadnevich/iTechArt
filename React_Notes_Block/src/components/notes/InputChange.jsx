import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import { Zoom } from '@material-ui/core';

import useStyles from './styled';

import { ButtonSave, ButtonDelete } from './ButtonReciever';

const InputChange = ({
  isChange,
  description,
  handleSaveNote,
  id,
  handleDelete,
  index,
  currentState,
}) => {
  const [updateDescription, setDescription] = useState(description);
  const { changeInputShow, changeInputHide, textAr } = useStyles();

  const sendDescription = (e) => {
    // eslint-disable-next-line no-unused-expressions
    e !== updateDescription && e ? setDescription(e) : description;
  };

  return (
    <Zoom in={isChange}>
      <Box className={isChange ? changeInputShow : changeInputHide}>
        <TextField
          className={textAr}
          label='Note Description'
          defaultValue={description}
          multiline
          rows={2}
          variant='standard'
          onChange={(e) => sendDescription(e.target.value)}
        />
        <Box>
          <ButtonSave
            updateDescription={updateDescription}
            handleSaveNote={handleSaveNote}
            id={id}
          />
          <ButtonDelete
            handleDelete={handleDelete}
            index={index}
            id={id}
            currentState={currentState}
          />
        </Box>
      </Box>
    </Zoom>
  );
};

export default InputChange;

InputChange.propTypes = {
  isChange: PropTypes.bool,
  handleSaveNote: PropTypes.func,
  description: PropTypes.string,
  id: PropTypes.number,
  index: PropTypes.number,
  handleDelete: PropTypes.func,
  currentState: PropTypes.string,
};

InputChange.defaultProps = {
  isChange: 'isChange',
  handleSaveNote: 'handleSaveNote',
  description: 'description',
  id: 'id',
  index: 'index',
  handleDelete: 'handleDelete',
  currentState: 'currentState',
};
