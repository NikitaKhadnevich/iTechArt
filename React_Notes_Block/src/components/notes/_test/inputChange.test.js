/* eslint-disable no-unused-vars */
/* eslint-disable no-shadow */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import * as reactRedux from 'react-redux';
import { configure, shallow, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import InputChange from '../InputChange';

configure({ adapter: new Adapter() }); // ES6 modules

describe('InputChange', () => {
  const inputChangeProps = {
    isChange: true,
    description: 'some descriprion',
    handleSaveNote: jest.fn(),
    id: 2,
    handleDelete: jest.fn(),
    index: 1,
    currentState: [
      {
        id: '',
        title: 'Some Title',
        description: 'Some description',
        date: '10/22/2021, 04:10:12 PM',
        isActive: false,
        isChange: false,
        isShared: false,
        sharedTime: '',
      },
    ],
  };

  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');

  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
  });

  it('Is it InputChange Form Exist', () => {
    const inputChanger = shallow(<InputChange {...inputChangeProps} />);
    expect(inputChanger).toMatchSnapshot();
    expect(inputChanger).toHaveLength(1);
  });
  it('Called Note Changer', () => {
    const inputChanger = shallow(<InputChange {...inputChangeProps} />);
    inputChangeProps.isChange = true;
    expect(inputChanger.find('ForwardRef(Zoom)').props().in).toBeTruthy();
  });
  it('InputChange Has Current Id and Index', () => {
    const inputChanger = shallow(<InputChange {...inputChangeProps} />);
    expect(inputChanger.find('ButtonSave').props().id).toBeTruthy();
    expect(inputChanger.find('ButtonDelete').props().id).toBeTruthy();
    expect(inputChanger.find('ButtonDelete').props().index).toBeTruthy();
  });
});
