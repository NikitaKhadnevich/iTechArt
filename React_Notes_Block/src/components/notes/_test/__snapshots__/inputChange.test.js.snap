// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputChange Is it InputChange Form Exist 1`] = `
<ForwardRef(Zoom)
  in={true}
>
  <Styled(ForwardRef(Box))>
    <Styled(ForwardRef(TextField))
      defaultValue="some descriprion"
      label="Note Description"
      multiline={true}
      onChange={[Function]}
      rows={2}
      variant="standard"
    />
    <ForwardRef(Box)>
      <ButtonSave
        handleSaveNote={[MockFunction]}
        id={2}
        updateDescription="some descriprion"
      />
      <ButtonDelete
        handleDelete={[MockFunction]}
        id={2}
        index={1}
      />
    </ForwardRef(Box)>
  </Styled(ForwardRef(Box))>
</ForwardRef(Zoom)>
`;
